extensions/amp-ad-network-cloudflare-impl/0.1/amp-ad-network-cloudflare-impl.js

```diff
@@ -64,16 +56,43 @@ export class AmpAdNetworkCloudflareImpl extends AmpA4A {
 
   /** @override */
   getAdUrl() {
-    // You can optionally use data from this context to augment the URL,
-    // to pass data back to the network, to aid in ad selection.  For example:
-    //   const rect = this.getIntersectionElementLayoutBox();
-    // can extract the layout size and be used to pass layout hints to the server.
+    const rect = this.getIntersectionElementLayoutBox();
+    const el = this.element;
+
+    const network = el.getAttribute('data-cf-network');
+    const base = NETWORKS[network].base;
+
+    // handle variable replacements
+    function replacements(str) {
+      if (!str) {
+        return str;
+      }
+
+      // allow injection of width and height as parameters
+      str = str.replace(/{width}/g, (rect.width || 0).toString());
```

Comment:
I expect that soon we'll want to allow arbitrary [analytics variable substitution](https://github.com/ampproject/amphtml/blob/master/spec/amp-var-substitutions.md) here. Would it be possible to change the syntax to `WIDTH` and `HEIGHT` without the braces, for compatibility with that? `SLOT_WIDTH` and `SLOT_HEIGHT` might be even better, to distinguish from other widths and heights, at the cost of slightly increased verbosity.